servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Completable.java

```diff
@@ -789,6 +789,19 @@ public final Completable afterOnSubscribe(Consumer<Cancellable> onSubscribe) {
         return afterSubscriber(doOnSubscribeSupplier(onSubscribe));
     }
 
+    /**
+     * Invokes the {@code onSubscribe} {@link Consumer} argument <strong>when</strong>
+     * {@link Subscriber#onSubscribe(Cancellable)} is called for {@link Subscriber}s of the returned
+     * {@link Completable}.
+     *
```

Comment:
(Applicable to all methods added in this PR) 

Following documents of `Publisher#whenOnSubscribe()`[1] can you also add the following:

```java
The order in which {@code onSubscribe} will be invoked relative to
{@link Subscriber#onSubscribe(Cancellable)} is undefined. If you need strict ordering see
{@link #beforeOnSubscribe(Consumer)} and {@link #afterOnSubscribe(Consumer)}.
```
Reference to the before/after methods

```java
     * @see #beforeOnNext(Consumer)
     * @see #afterOnNext(Consumer)
```
[1]: https://github.com/apple/servicetalk/blob/master/servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Publisher.java#L562-L576

Reply:
Thanks for the review. I have updated the documentations for the methods `whenOnSubscribe`: but I am not sure what to write for `whenSubscriber` method which have arguments like `(Supplier<? extends Subscriber<? super T>> subscriberSupplier)`.

Should it be same ? 

```
     * <p>
     * The order in which {@code onSubscribe} will be invoked relative to
     * {@link Subscriber#onSubscribe(Cancellable)} is undefined. If you need strict ordering see
     * {@link #beforeOnSubscribe(Consumer)} and {@link #afterOnSubscribe(Consumer)}.

```

and 

```
     * @see #beforeOnNext(Consumer)
     * @see #afterOnNext(Consumer)
```

Reply:
@NiteshKant please review the changes.

Reply:
I have commented on the changes for `Completable` which can now be replicated to other classes also. LMK if you have questions.

Reply:
Thanks for the review @NiteshKant ! I have done the changes.